### PR TITLE
Issue #95 datetime picker

### PR DESCRIPTION
**Pull request by Agri**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -70,17 +70,21 @@
     "webpack-merge": "^4.1.2"
   },
   "dependencies": {
+    "@types/moment": "^2.13.0",
```
Comment (sharpcoding) [inline package.json]: @types dependencies should go to "devDependencies"

File: src/components/datepicker/datepickerwrapper.tsx [hunk 2]
```diff
@@ -0,0 +1,93 @@
+import * as React from 'react';
+import * as moment from 'moment';
+import { connect } from 'react-redux';
+import DatePicker from 'react-datepicker';
+import { IState } from '../../state';
+import { bindActionCreators, Dispatch } from 'redux';
```
Comment (sharpcoding) [inline src/components/datepicker/datepickerwrapper.tsx]: order of imports violated, please use ts-lint with vs-code

File: src/components/datepicker/datepickerwrapper.tsx [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+interface IDatePickerWrapperComponentProps {
+
+}
+
+interface IDatePickerWrapperComponentActionCreators {
+    dateSelected: (startDate: string, endDate: string) => void;
+}
+
+interface IDatePickerWrapperComponentState {
+    startDate: moment.Moment;
```
Comment (sharpcoding) [inline src/components/datepicker/datepickerwrapper.tsx]: - there is no agreement on how date should be stored
- we don't plan to use momentjs
the best solution would be using ```Date``` object

File: src/components/datepicker/datepickerwrapper.tsx [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+    endDate: moment.Moment;
+}
+
+export class DatePickerWrapper extends React.Component<IDatePickerWrapperComponentProps & IDatePickerWrapperComponentActionCreators, IDatePickerWrapperComponentState> {
+    private dateFormat: string = 'YYYY-MM-DDTHH:mm';
+    constructor(props: IDatePickerWrapperComponentProps & IDatePickerWrapperComponentActionCreators, context: any) {
+        super(props, context);
+        this.state = {
+            startDate: moment().hours(0).minutes(0),
```
Comment (sharpcoding) [inline src/components/datepicker/datepickerwrapper.tsx]: as above - don't use moment.js

Comment (Agri) [reply src/components/datepicker/datepickerwrapper.tsx]: this ReactDatePicker is fully based on moment.js. Should I do research to find another one that dont use it?

Comment (sharpcoding) [reply src/components/datepicker/datepickerwrapper.tsx]: Yes, please use https://www.npmjs.com/package/react-bootstrap-date-picker

Comment (sharpcoding) [reply src/components/datepicker/datepickerwrapper.tsx]: I found another bug in react-datepicker package, so le't forget about this control for a while

Comment (Agri) [reply src/components/datepicker/datepickerwrapper.tsx]: This control doesn't want to work in anomalies environment. Or plz send working piece of code. 
image

File: src/components/datepicker/datepickerwrapper.tsx [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+    } as IDatePickerWrapperComponentProps;
+}
+
+function matchDispatchToProps(dispatch: Dispatch<{}>) {
+    return bindActionCreators({
+
+    }, dispatch);
+}
+
+export default connect(mapStateToProps, matchDispatchToProps)(DatePickerWrapper);
```
Comment (sharpcoding) [inline src/components/datepicker/datepickerwrapper.tsx]: Sorry, no connect at the level of a simple representational component, more on this: https://medium.com/@dan_abramov/smart-and-dumb-components-7ca2f9a7c7d0

File: src/components/datepicker/datepickerwrapper.tsx (hunk 4 above)
Comment (sharpcoding) [reply src/components/datepicker/datepickerwrapper.tsx]: Yes, it does not make any sense to advocate for react-bootstrap-date-picker

File: src/components/datepicker/date-picker.tsx [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+}
+
+export default class DatePickerWrapper extends React.Component<IDatePickerWrapperComponentProps & IDatePickerWrapperComponentActionCreators, IDatePickerWrapperComponentState> {
+    private dateFormat: string = 'YYYY-MM-DDTHH:mm';
+    constructor(props: IDatePickerWrapperComponentProps & IDatePickerWrapperComponentActionCreators, context: any) {
+        super(props, context);
+        this.state = {
+            startDate: moment().hours(0).minutes(0),
+            endDate: moment().hours(0).minutes(0),
+        };
```
Comment (tpichlak) [inline src/components/datepicker/date-picker.tsx]: This initial startDate and endDate should be different, here is an example:
```
const startOfDay = moment().startOf('day');
        this.state = {
            startDate: startOfDay.clone().subtract(3, 'months'),
            endDate: startOfDay,
        };
```

File: src/components/datepicker/date-picker.tsx [hunk 7]
```diff
@@ -0,0 +1,76 @@
+import * as moment from 'moment';
+import * as React from 'react';
+import { ControlLabel, Form, FormGroup } from 'react-bootstrap';
+import DatePicker from 'react-datepicker';
+import 'react-datepicker/dist/react-datepicker-cssmodules.css';
+import 'react-datepicker/dist/react-datepicker.css';
+
+interface IDatePickerWrapperComponentProps {
+
+}
```
Comment (tpichlak) [inline src/components/datepicker/date-picker.tsx]: Remove empty interface from code.

**Pull request merged**